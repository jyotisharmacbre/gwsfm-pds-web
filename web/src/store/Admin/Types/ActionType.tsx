export enum ActionType {
	GET_PROJECT_PARAMETERS_SUCCESS = 'GET_PROJECT_PARAMETERS_SUCCESS',
	GET_PROJECT_PARAMETERS_ERROR = 'GET_PROJECT_PARAMETERS_ERROR'
}
