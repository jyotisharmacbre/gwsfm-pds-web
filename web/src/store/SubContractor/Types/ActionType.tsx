export enum ActionType {
	SUB_CONTRACTOR_FORM_ADD_SUCCESS = 'SUB_CONTRACTOR_FORM_ADD_SUCCESS',
	SUB_CONTRACTOR_FORM_EDIT_SUCCESS = 'SUB_CONTRACTOR_FORM_EDIT_SUCCESS',
	SUB_CONTRACTOR_FORM_ERROR = 'SUB_CONTRACTOR_FORM_ERROR',
	GET_SUB_CONTRACTOR_SUCCESS = 'GET_SUB_CONTRACTOR_SUCCESS',
	GET_SUB_CONTRACTOR_ERROR = 'GET_SUB_CONTRACTOR_ERROR',
	RESET_SUB_CONTRACTOR_STATE = 'RESET_SUB_CONTRACTOR_STATE',
	RESET_SUB_CONTRACTOR_NOTIFIER = 'RESET_SUB_CONTRACTOR_NOTIFIER'
}
