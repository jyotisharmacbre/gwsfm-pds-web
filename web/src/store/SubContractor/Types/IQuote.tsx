export interface IQuote {
   activityQuoteId: string;
        subContrActivityId: string;
        supplierName: string;
        quoteValue: 0;
}
