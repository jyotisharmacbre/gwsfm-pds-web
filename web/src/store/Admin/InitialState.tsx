import { IAdminState } from './Types/IAdminState';
import { IAdminDefaults } from './Types/IAdminDefault';

export const initialState: IAdminState = {
	adminDefaultValues: [],
	error: ''
};
