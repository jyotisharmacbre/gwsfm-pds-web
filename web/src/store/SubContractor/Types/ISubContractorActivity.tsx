import {IQuote} from './IQuote';

export interface ISubContractorActivity {
  activityName: string;
  existingSubcontractor: boolean;
  subcontractor: string;
  preferredSupplier: boolean;
  totalCost: number;
  grossMargin: number;
  totalSell: number;
  comments: string;
  quote1:IQuote;
  quote2:IQuote;
  quote3:IQuote;
}
