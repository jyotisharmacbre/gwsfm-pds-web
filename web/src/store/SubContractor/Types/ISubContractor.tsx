import { ISubContractorActivity } from './ISubContractorActivity';
export interface ISubContractor {
  activities: Array<ISubContractorActivity>;
}
