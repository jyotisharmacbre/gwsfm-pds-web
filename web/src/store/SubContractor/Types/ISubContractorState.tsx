import { ISubContractor } from './ISubContractor';
import Notify from '../../../enums/Notify';
import EventType from '../../../enums/EventType';

export interface ISubContractorState {
  form: ISubContractor;
  error: string | null;
  loading: boolean;
  notify: Notify;
  event: EventType;
}
