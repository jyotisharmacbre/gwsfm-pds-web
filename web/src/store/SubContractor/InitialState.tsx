import Notify from '../../enums/Notify';
import EventType from '../../enums/EventType';
import {ISubContractorActivity} from './Types/ISubContractorActivity';
import { ISubContractorState } from './Types/ISubContractorState';
import {IQuote} from './Types/IQuote';

export const newQuote : IQuote = {
     activityQuoteId: '',
        subContrActivityId: '',
        supplierName: '',
        quoteValue: 0
};

export const newActivity : ISubContractorActivity ={
    subContrActivityId: '',
    projectId: '',
    activityName: '', 
    isExistingSubcontractor: null,
    subcontractorId: '',
    isPreferredSupplier: null,
    totalCost: 0,
    grossMargin: 0,
    totalSell: 0,
    comments: '',
    quotes:[
      {...newQuote},
      {...newQuote},
      {...newQuote}
      ]
};

export const initialState: ISubContractorState = {
  form: {
    activities: [{...newActivity}]
  },
  error: null,
  loading: false,
  notify: Notify.none,
  event: EventType.none
};