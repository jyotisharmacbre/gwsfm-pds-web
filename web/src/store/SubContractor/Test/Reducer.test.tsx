import React from 'react';
import {ActionType} from '../Types/ActionType';
import subContractorReducer from '../Reducer';
import { initialState } from '../InitialState';
import nock from 'nock';
import { baseURL } from '../../../client/client';

  nock(baseURL)
  .post('/api/SubContractor/activities')
  .reply(200, "Sub oontractor form added successfully");

  describe('Sub Contractor Reducer', () => {
      it('should handle add sub contractor successfully', () => {
        const subContractorFormAddAction: any = {
          type: ActionType.SUB_CONTRACTOR_FORM_ADD_SUCCESS
        };
        expect(
          subContractorReducer(initialState, subContractorFormAddAction)
        ).toMatchSnapshot();
      });   
});
