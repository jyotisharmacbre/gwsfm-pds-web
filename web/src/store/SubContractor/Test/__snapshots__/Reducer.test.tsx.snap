// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sub Contractor Reducer should handle add new activity 1`] = `
Object {
  "error": null,
  "event": 3,
  "form": Object {
    "activities": Array [
      Object {
        "activityName": "",
        "comments": "",
        "grossMargin": 0,
        "isExistingSubcontractor": null,
        "isPreferredSupplier": null,
        "projectId": "",
        "quotes": Array [
          Object {
            "activityQuoteId": "",
            "quoteValue": 0,
            "subContrActivityId": "",
            "supplierName": "",
          },
          Object {
            "activityQuoteId": "",
            "quoteValue": 0,
            "subContrActivityId": "",
            "supplierName": "",
          },
          Object {
            "activityQuoteId": "",
            "quoteValue": 0,
            "subContrActivityId": "",
            "supplierName": "",
          },
        ],
        "subContrActivityId": "",
        "subcontractorId": "",
        "totalCost": 0,
        "totalSell": 0,
      },
    ],
  },
  "loading": false,
  "notify": 0,
}
`;

exports[`Sub Contractor Reducer should handle add sub contractor successfully 1`] = `
Object {
  "error": null,
  "event": undefined,
  "form": Object {
    "activities": Array [
      Object {
        "activityName": "",
        "comments": "",
        "grossMargin": 0,
        "isExistingSubcontractor": null,
        "isPreferredSupplier": null,
        "projectId": "",
        "quotes": Array [
          Object {
            "activityQuoteId": "",
            "quoteValue": 0,
            "subContrActivityId": "",
            "supplierName": "",
          },
          Object {
            "activityQuoteId": "",
            "quoteValue": 0,
            "subContrActivityId": "",
            "supplierName": "",
          },
          Object {
            "activityQuoteId": "",
            "quoteValue": 0,
            "subContrActivityId": "",
            "supplierName": "",
          },
        ],
        "subContrActivityId": "",
        "subcontractorId": "",
        "totalCost": 0,
        "totalSell": 0,
      },
    ],
  },
  "loading": false,
  "notify": 1,
}
`;
